### PR TITLE
MULTISITE-16110 - Allow empty file site.make

### PR DESCRIPTION
**Pull request by jonhy81**
The idea behind this approach is to drop the file ```site.make.example``` and allow to run the target ```build-dist``` with an empty ```site.make``` file.

### CODE REVIEW

File: build.package.xml [hunk 1]
```diff
@@ -126,10 +126,26 @@
...
             <available file="${subsite.make}" type="file" />
             <then>
-                <phingcall target="drush-make-no-core">
-                    <property name="drush.make.target.file" value="${subsite.make}" />
-                    <property name="drush.make.root" value="${phing.project.tmp.dir}/build" />
-                </phingcall>
+                <loadfile property="sitemake" file="${subsite.make}"/>
+                <propertyregex property="not.empty"
+                               subject="${sitemake}"
+                               pattern="(libraries\[|projects\[)"
```
Comment (verbruggenalex) [inline build.package.xml]: Good idea, at first I thought regex wasnt possible, but apparently it works.
We need to take commented lines into account as well. So this regex should look for lines not starting with # or ;

File: build.package.xml [hunk 2]
```diff
@@ -126,10 +126,26 @@
...
+                               match="$1"
+                               casesensitive="false"
```
Comment (verbruggenalex) [inline build.package.xml]: I tested capital, like PROJECTS and the make fifle doesn't build. So I think we should make the regex case sensitive.

File: build.package.xml [hunk 3]
```diff
@@ -34,10 +34,10 @@
             <then>
                 <echo msg='rm -rf "${folder.to.delete}"' />
                 <exec
-                    command='rm -rf "${folder.to.delete}"'
```
Comment (verbruggenalex) [inline build.package.xml]: Let's  discuss on implementing a phpcs standard on these build xml files, cause this happens a lot.

**Comment by verbruggenalex** [comment]
Thinking about this a bit more. Regex is not the way to go.
- We just need to perform a try catch on the drush make execution.
- And a check to see if directory exists before we try to move it.

**Comment by jonhy81** [comment]
To do a try catch in drush make command the best way is to code to drush that is out of our scope. Maintain drush in our side is not a good idea.

**Pull request merged**